src/hooks/useConversations.ts

```diff
@@ -0,0 +1,28 @@
+import { useMemo } from 'react'
+import { sortDateDesc } from '../helpers'
+import useChat from './useChat'
+
+const useConversations = () => {
+  const { store } = useChat()
+
+  const peers = useMemo(() => Object.keys(store), [store])
+  const peersWithLatestDate = useMemo(
+    () =>
+      peers.map((peerAddress) => {
+        const messages = store[peerAddress] || []
+        const lastMessage = messages[messages.length - 1]
+        return {
+          peerAddress,
+          sent: lastMessage.sent,
+        }
```

Comment:
I think this would result in an error if `messages` was an empty array. Because `lastMessage` would be `undefined` and so `lastMessage.sent` would not work. Probably just need to return with `sent: lastMessage?.sent`

Reply:
[6c9d25c](https://github.com/liteflow-labs/chat/pull/15/commits/6c9d25c42e80711f54005bac7106d07f1317f669)